source/main.cpp

```diff
@@ -289,6 +289,18 @@ namespace Aws
 
 int main(int argc, char *argv[])
 {
+    auto listener = std::make_shared<DefaultClientBaseNotifier>();
```

Comment:
Currently `SharedCrtResourceManager::initialize` does 3 things.
1. Stores a reference to the feature registry 
2. Initializes the allocator (`SharedCrtResourceManager::initializeAllocator`)
3. Initializes the SDK client handle (`SharedCrtResourceManager::buildClient`)

Step 3 requires that the `Config` passed to `SharedCrtResourceManager::initialize` be initialized with the `CliArgs` **before** calling this function whereas the proposed change on this commit initializes `Config`  from `CliArgs` **after** calling `SharedCrtResourceManager::initialize`.

In order to avoid breaking this invariant, I recommend that you make the following changes to this commit.

* Make the function to initialize the allocator public (`SharedCrtResourceManager::initializeAllocator`) and remove the `Config` from the signature since it actually isn't used to initialize the allocator. 
* Remove the call to `SharedCrtResourceManager::initializeAllocator` from `SharedCrtResourceManager::initialize`.
* Create the shared instance to `resourceManager` as the first operation in `main` and invoke the `SharedCrtResourceManager::initializeAllocator`.

The other changes in this commit aren't required.